collectors/proc.plugin/sys_class_infiniband.c

```diff
@@ -540,7 +540,8 @@ int do_sys_class_infiniband(int update_every, usec_t dt)
                 rrdset_flag_set(port->st_bytes, RRDSET_FLAG_DETAIL);
                 // On this chart, we want to have a KB/s so the dashboard will autoscale it
                 // The reported values are also per-lane, so we must multiply it by the width
-                FOREACH_COUNTER_BYTES(GEN_RRD_DIM_ADD_CUSTOM, port, 8 * port->width, 1024, RRD_ALGORITHM_INCREMENTAL)
+                // x4 lanes multiplier as per Documentation/ABI/stable/sysfs-class-infiniband
+                FOREACH_COUNTER_BYTES(GEN_RRD_DIM_ADD_CUSTOM, port, 4 * 8 * port->width, 1024, RRD_ALGORITHM_INCREMENTAL)
```

Comment:
There is a problem when converting units from A to B using multiplier/divider: prometheus exporting (perhaps not only) with "as-collected"  option will be broken.

Reply:
Found related topic: https://community.netdata.cloud/t/prometheus-output-reports-disk-usage-differently-than-netdata/1139/2

Reply:
Would you prefer that this multiplication is done for each reading on the value, rather than at dimension definition ?

Reply:
The general advice is to do all the calculations/transformations in the code and using divisor/multiplier only if you need a value to be negative (representation).